### PR TITLE
Packages: Fix components and search storybook

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR fixes storybook instances of the components and search packages. Not sure how it worked before, but running storybook now errors out with some missing modules (`ts-loader` and `react-docgen-typescript-loader`). So this PR adds those in custom storybook webpack configs. In addition, we remove a `emitDeclarationOnly` and simplify the way we do the mobile viewport check, to avoid extra complications from importing `@automattic/viewport`, `@wordpress/viewport` or `useViewportMatch` from `@wordpress/compose`.

Aims to unblock #46825.

#### Testing instructions

* Run `yarn`
* Run `yarn run components:storybook:start` and verify it builds and works
* Run `yarn run search:storybook:start`  and verify it builds and works

#### Next steps

I think we can improve the webpack configs a bit to improve build performance, and introduce some caching, but this should do the job for now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -289,6 +289,7 @@
 		"qs": "^6.9.1",
 		"react": "^16.12.0",
 		"react-dom": "^16.12.0",
+		"react-docgen-typescript-loader": "^3.7.2",
```
Comment (scinos) [inline package.json:292]: Can we move these new deps to `packages/components/package.json`?

Comment (tyxla) [reply package.json:292]: Sure!

File: packages/components/.storybook/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const path = require( 'path' );
+
+module.exports = ( { config } ) => {
+	config.module.rules.push( {
+		test: /\.tsx?$/,
+		include: path.resolve( __dirname, '../src' ),
```
Comment (scinos) [inline packages/components/.storybook/webpack.config.js:6]: nit: `path.join(__dirname, '../src')` does the same and arguably is more correct.

File: packages/components/.storybook/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		use: [
+			require.resolve( 'ts-loader' ),
```
Comment (scinos) [inline packages/components/.storybook/webpack.config.js:8]: Isn't this the same as just passing `ts-loader`? Why do the resolution here?

File: yarn.lock [hunk 4]
```diff
@@ -11741,19 +11753,19 @@ enhanced-resolve@4.1.0:
...
   dependencies:
     graceful-fs "^4.1.2"
     memory-fs "^0.5.0"
     tapable "^1.0.0"
 
-enhanced-resolve@^4.1.1, enhanced-resolve@^4.3.0:
-  version "4.3.0"
-  resolved "https://registry.yarnpkg.com/enhanced-resolve/-/enhanced-resolve-4.3.0.tgz#3b806f3bfafc1ec7de69551ef93cca46c1704126"
-  integrity sha512-3e87LvavsdxyoCfGusJnrZ5G8SLPOFeHSNpZI/ATL9a5leXo2k0w6MKnbqhdBad9qTobSfB20Ld7UmgoNbAZkQ==
+enhanced-resolve@^4.1.0:
```
Comment (scinos) [inline yarn.lock:11765]: Can be deduped to 4.3.0

**Review by scinos — CHANGES_REQUESTED** [review]
I think the new dependencies should be declared in the package.json of the package using them, not in the root.

The rest are nitpicks.

File: package.json (hunk 1 above)
Comment (tyxla) [reply package.json]: Done in a1d3bb479ec1f0e4178956e0def059ecdbfb031a

File: packages/components/.storybook/webpack.config.js (hunk 2 above)
Comment (tyxla) [reply packages/components/.storybook/webpack.config.js]: Good call, done in bef964f1d8b76dc3c655ab5aa0eb977af62e0da0

File: packages/components/.storybook/webpack.config.js (hunk 3 above)
Comment (tyxla) [reply packages/components/.storybook/webpack.config.js]: 🤷  wrote it in "hack" mode. Good call, fixed in bef964f1d8b76dc3c655ab5aa0eb977af62e0da0

File: yarn.lock (hunk 4 above)
Comment (tyxla) [reply yarn.lock]: Deduped in aafe8ead2eb6c945400573b4f7b5fbeade460956

**Comment by tyxla** [comment]
@scinos thanks for the swift review! I've addressed the feedback, feel free to take another look.

**Review by scinos — APPROVED** [review]

**Pull request merged**